### PR TITLE
Java JSON Unicode Support

### PR DESCRIPTION
**Pull request by TanayParikh**
- Added a HexCharParser
- Added a UnicodeCharParser
- Added a ASCIIEscapeParser (Supports `\n, \t, \r, \f, \b`)
- Added Unicode parsing support to the Java String parser
- Added Unicode parsing support for JS JSON parser

### CODE REVIEW

File: src/foam/lib/json/UnicodeParser.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  public PStream parse(PStream ps, ParserContext x) {
+    ps = super.parse(ps, x);
+
+    if ( ps != null ) {
+      Object[] values = (Object[]) ps.value();
+
+      // Gets the repeated hex char sequence
+      values = (Object[]) values[3];
+
+      char hexChar =  ( (char)
```
Comment (kgrgreer) [inline src/foam/lib/json/UnicodeParser.java]: Extra parens not needed.

File: src/foam/lib/json/UnicodeParser.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+    return ps;
+  }
+
+  public static int hexToInt(Object o, int power) {
+    char c = o.toString().charAt(0);
+
+    // Converts to upper case if necessary
+    if ( c >= 'a' && c <= 'z' ) {
+        c = Character.toUpperCase(c);
```
Comment (kgrgreer) [inline src/foam/lib/json/UnicodeParser.java]: indent 2 spaces

Comment (TanayParikh) [reply src/foam/lib/json/UnicodeParser.java]: Updated.

File: src/foam/lib/json/UnicodeParser.java (hunk 1 above)
Comment (TanayParikh) [reply src/foam/lib/json/UnicodeParser.java]: Removed.

**Pull request merged**